app/models/user.rb

```diff
@@ -1,18 +1,22 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable, :registerable, and :omniauthable
   devise :rememberable, :omniauthable, omniauth_providers: [:github]
 
-  validates :login, :uid, presence: true, uniqueness: true
-  validates :email, :uid, presence: true, uniqueness: true
+  validates :login, :email, :uid, presence: true, uniqueness: true
 
   def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first_or_create(
-      email:    auth.info.email,
-      name:     auth.info.name,
-      login:    auth.extra.raw_info.login
+    user = where(provider: auth.provider, uid: auth.uid).first_or_initialize(
+      email:  auth.info.email,
+      name:   auth.info.name,
+      login:  auth.extra.raw_info.login
     )
+    user.token = auth.credentials.token
+    user
+  end
+
+  def organisations
+    user = Github.new oauth_token: token, auto_pagination: true
+    user.orgs.list.body.map { |org| org[:login]&.downcase }.compact
```

Comment:
Please add a Ruby comment here, pointing to Github API documentation/your dependency source code explaining why this is never nil, and always an empty array — and I'll makr this COdeClimate as invalid.